### PR TITLE
Arnikola/subfunctions

### PR DESCRIPTION
**Pull request by arnikola**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/block/reconsolidated.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	next := it.it.Next()
+	it.ts = it.ts.Add(it.it.Meta().Bounds.StepSize)
+	return next
+}
+
+func (it *ucReconsolidatedStepIter) Current() UnconsolidatedStep {
+	var (
+		c      = it.it.Current()
+		values = c.Values()
+		dpList = make([]ts.Datapoints, 0, len(values))
```
Comment (robskillington) [inline src/query/block/reconsolidated.go]: Guess there's no way to reuse these?

File: src/query/block/reconsolidated.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	next := it.it.Next()
+	it.idx = it.idx + 1
+	return next
+}
+
+func (it *ucReconsolidatedSeriesIter) Current() UnconsolidatedSeries {
+	var (
+		c      = it.it.Current()
+		values = c.values
+		dpList = make([]ts.Datapoints, 0, len(values))
```
Comment (robskillington) [inline src/query/block/reconsolidated.go]: Similarly here, no way to reuse these by any chance?

File: src/query/block/reconsolidated.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	return b.block.StepIter()
+}
+
+// SeriesIter returns a SeriesIterator
+func (b *reconsolidatedBlock) SeriesIter() (SeriesIter, error) {
+	return b.block.SeriesIter()
+}
+
+// Unconsolidated returns the unconsolidated version for the block
+func (b *reconsolidatedBlock) Unconsolidated() (UnconsolidatedBlock, error) {
```
Comment (robskillington) [inline src/query/block/reconsolidated.go]: Hm, wonder if we should come up with some terms for this.  `reconsolidatedBlock` and `ucReconsolidatedBlock` definitely took me a minute or so to totally grok the relationship, etc just based on name.  I get it know, but yeah, hehe.

File: src/query/models/types.go [hunk 4]
```diff
@@ -99,6 +100,14 @@ type TagOptions interface {
 	IDSchemeType() IDSchemeType
 }
 
+type FetchOverrideOptions struct {
+	ShouldOverride bool
```
Comment (robskillington) [inline src/query/models/types.go]: Hehe, I like the avoiding the heap allocation by using a bool instead of making it a pointer type (for optionality).

Comment (robskillington) [reply src/query/models/types.go]: I really wish Go just had bloody optionals (i.e. unwrapping) baked in to the language in a zero alloc way.

**Comment by arnikola** [comment]
Deprecated in favour of https://github.com/m3db/m3/pull/2143 (which itself is quite old and possibly deprecated)

**Pull request closed without merging**